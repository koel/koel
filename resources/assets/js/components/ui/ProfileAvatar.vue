<template>
  <a
    v-koel-tooltip.left
    class="view-profile"
    data-testid="view-profile-link"
    href="/#/profile"
    title="Profile and preferences"
  >
    <img :alt="`Avatar of ${currentUser.name}`" :src="currentUser.avatar"/>
  </a>
</template>

<script lang="ts" setup>
import { useAuthorization } from '@/composables'

const { currentUser } = useAuthorization()
</script>

<style lang="scss" scoped>
img {
  display: block;
  width: 39px;
  aspect-ratio: 1/1;
  border-radius: 50%;
  padding: 2px;
  border: 1px solid rgba(255, 255, 255, .1);
  transition: border .2s ease-in-out;

  &:hover {
    border-color: rgba(255, 255, 255, .3);
  }

  &:active {
    transform: scale(.95);
  }
}
</style>
