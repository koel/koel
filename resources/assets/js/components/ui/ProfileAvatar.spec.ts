import { expect, it } from 'vitest'
import factory from '@/__tests__/factory'
import UnitTestCase from '@/__tests__/UnitTestCase'
import ProfileAvatar from './ProfileAvatar.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders', () => {
      const user = factory<User>('user', {
        name: 'John Doe',
        avatar: 'https://example.com/avatar.jpg'
      })

      expect(this.actingAs(user).render(ProfileAvatar).html()).toMatchSnapshot()
    })
  }
}
