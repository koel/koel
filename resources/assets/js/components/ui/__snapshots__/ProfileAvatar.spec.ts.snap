// Vitest Snapshot v1

exports[`renders 1`] = `<a class="view-profile" data-testid="view-profile-link" href="/#/profile" title="Profile and preferences" data-v-663f2e50=""><img alt="Avatar of John Doe" src="https://example.com/avatar.jpg" data-v-663f2e50=""></a>`;
